src/Castle.Core/DynamicProxy/DefaultProxyBuilder.cs

```diff
@@ -78,6 +78,14 @@ public Type CreateClassProxyType(Type classToProxy, Type[] additionalInterfacesT
 			return generator.GetGeneratedType();
 		}
 
+		public Type CreateDelegateProxyType(Type delegateToProxy)
+		{
+			AssertValidType(delegateToProxy);
```

Comment:
As an aside, I noticed a fair amount of code duplication in the methods used for validating arguments.

* For `DefaultProxyBuilder`, there's `AssertValidType` and `AssertValidTypes`.
* For `ProxyGenerator`, there is `CheckNotGenericTypeDefinition` and `CheckNotGenericTypeDefinitions`.
* In `ProxyGenerator`, I opted not to use the latter because they throw the wrong type of exception (`GeneratorException` when I needed `ArgumentException`), so I duplicated the same validation logic once again.

These all do very similar things. Perhaps we could merge them sometime.

Reply:
I'm interpreting your :+1: as, "Yes, let's get rid of this code duplication (but in a separate PR)." Let me know if I misinterpreted. :)